### PR TITLE
feat(barchart)!: allow axes to accept Lines

### PR DESCRIPTION
**Pull request by joshka**
Fixes: https://github.com/ratatui-org/ratatui/issues/1272

### CODE REVIEW

File: src/widgets/chart.rs [hunk 1]
```diff
@@ -904,17 +908,13 @@ impl<'a> Chart<'a> {
...
-
-        let x = match alignment {
-            Alignment::Left => label_area.left(),
-            Alignment::Center => label_area.left() + label_area.width / 2 - bounded_label_width / 2,
-            Alignment::Right => label_area.right() - bounded_label_width,
+    fn render_label(buf: &mut Buffer, label: &Line, label_area: Rect, alignment: Alignment) {
+        let label = match alignment {
+            Alignment::Left => label.clone().left_aligned(),
+            Alignment::Center => label.clone().centered(),
+            Alignment::Right => label.clone().right_aligned(),
```
Comment (joshka) [inline src/widgets/chart.rs:915]: There are usually a fairly small number of short labels here, so the clone is probably ok here.

File: tests/widgets_chart.rs [hunk 2]
```diff
@@ -203,7 +203,7 @@ fn widgets_chart_handles_long_labels<'line, Lines>(
         "          ",
         "          ",
         "──────────",
-        "AAA  B   C",
+        "AAA B    C",
```
Comment (joshka) [inline tests/widgets_chart.rs:206]: This is better aligned with how line treats centering (ambiguous centering preferences to the left)

**Comment by joshka** [comment]
Dependencies action is failing due to docker build.

**Review by orhun — APPROVED** [review]

File: src/widgets/chart.rs (hunk 1 above)
Comment (EdJoPaTo) [reply src/widgets/chart.rs:915]: Why is it needed in the first place? :thinking: do the alignment methods require ownership?

File: src/widgets/chart.rs [hunk 3]
```diff
@@ -97,8 +101,8 @@ impl<'a> Axis<'a> {
     ///         .labels(vec!["0".bold(), "25".into(), "50".bold()]);
     /// ```
     #[must_use = "method moves the value of self and returns the modified value"]
-    pub fn labels(mut self, labels: Vec<Span<'a>>) -> Self {
-        self.labels = Some(labels);
+    pub fn labels<T: Into<Line<'a>>>(mut self, labels: Vec<T>) -> Self {
+        self.labels = Some(labels.into_iter().map(Into::into).collect());
```
Comment (EdJoPaTo) [inline src/widgets/chart.rs:105]: Why not `IntoIterator` then too? That would allow for `.labels(["0.0", "5.0", "10.0"])` without `vec![]`.

Comment (joshka) [reply src/widgets/chart.rs:105]: Sounds good - want to slap together a quick PR for that?

File: src/widgets/chart.rs (hunk 1 above)
Comment (EdJoPaTo) [reply src/widgets/chart.rs:915]: Ah, it moves the original… well…

**Pull request merged**